appinventor/blocklyeditor/src/workspace_svg.js

```diff
@@ -1012,6 +1012,30 @@ Blockly.WorkspaceSvg.prototype.customContextMenu = function(menuOptions) {
   };
   menuOptions.push(enableAll);
 
+  // Clear unused blocks
+  var clearUnusedBlocks = {enabled: true};
+  clearUnusedBlocks.text = Blockly.Msg.REMOVE_UNUSED_BLOCKS;
+  clearUnusedBlocks.callback = function() {
+    var response = confirm(Blockly.Msg.CONFIRM_DELETE)
```

Comment:
I'm not sure that this is the logic we want here.I think what we should do is compute the list of non-declaration top level blocks and present the dialog with "Are you sure you want to remove N blocks?" so the user is aware how many blocks will be trashed first. The deletion operations should be wrapped in an event group so if the user presses ctrl+z they will get them all back.

Note that the only valid declaration blocks are global declarations, procedure definitions, and event handler blocks

Reply:
@ewpatton sir just to summarize  we have ro remove non declaration blocks (for example: if-else block) and top level blocks (those blocks that does not have any parent block and no child blocks ) right?

Reply:
@ewpatton sir I have made changes as you said. here is what i have done 
`if(block.previousConnection != null || (block.outputConnection && block.outputConnection != null)) {
          removeList.push(block)
      }`
this makes sure that the block is of non-declarative type. after that i have opened a dialog with a message 
"Are you sure you want to delete all X of these blocks?" and if user clicks ok it will delete those.

In addition to that i have also wrapped as event group so that undo functionality works

Reply:
@ewpatton sir i have made changes as you said can you please review it so that i can work on other issues?

Reply:
@ewpatton Can you please check the latest commit I have made? I did changes as per your inputs